ios/browser/api/history/brave_history_api.h

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_IOS_BROWSER_API_HISTORY_BRAVE_HISTORY_API_H_
+#define BRAVE_IOS_BROWSER_API_HISTORY_BRAVE_HISTORY_API_H_
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSUInteger, BraveHistoryTransitionType) {
+  BraveHistoryTransitionType_LINK,
+  BraveHistoryTransitionType_TYPED,
```

Comment:
see https://google.github.io/styleguide/objcguide.html#constants for enum naming, but this doesn't cover all the options and we need a method to convert between ui::PageTransition and PageTransitionIOS

Reply:
This enum is changed with a boolean. Mobile side we only have the information If the url is typed or not.